### PR TITLE
EVG-2548 Add integration tests to new ec2 host provider

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: cloud/new_ec2.go [hunk 1]
```diff
@@ -139,19 +139,21 @@ func (m *ec2Manager) spawnOnDemandHost(h *host.Host, ec2Settings *NewEC2Provider
...
 		grip.Error(message.WrapError(h.Remove(), message.Fields{
-			"message":       "error removing intent host",
-			"host":          h.Id,
-			"host_provider": h.Distro.Provider,
-			"distro":        h.Distro.Id,
+			"cloud_provider": "ec2",
+			"message":        "error removing intent host",
+			"host":           h.Id,
+			"host_provider":  h.Distro.Provider,
+			"distro":         h.Distro.Id,
```
Comment (tychoish) [inline cloud/new_ec2.go]: we don't get the reason why we're removing the intent host here, (either err, or reservation,) so this is only logged if we have trouble with the database connection.

also cloud provider and host provider are probably always mostly the same thing? or at least cloud provider won't add much information.

Comment (tychoish) [reply cloud/new_ec2.go]: does h.Remove() set h to nil?

File: cloud/new_ec2.go [hunk 2]
```diff
@@ -162,28 +164,31 @@ func (m *ec2Manager) spawnOnDemandHost(h *host.Host, ec2Settings *NewEC2Provider
 
 	instance := reservation.Instances[0]
 	grip.Debug(message.Fields{
-		"message": "started ec2 instance",
-		"host":    *instance.InstanceId,
-		"distro":  h.Distro.Id,
+		"cloud_provider": "ec2",
```
Comment (tychoish) [inline cloud/new_ec2.go]: use h.distro.provider rathre than string.

File: cloud/new_ec2_integration_test.go [hunk 3]
```diff
@@ -12,22 +12,52 @@ import (
...
+		Teardown:    "",
+		User:        "root",
+		SSHKey:      "",
+	}
+}
+
+// This spawns (and terminates) an actual on-demand instance in AWS. Since AWS bills for a minimum
+// of 1 minute, this will cost approximately $.0002 each time it is run (t1.micro
+// $0.0116 per hour / 60 minutes). Since the price is low, we run this each time
+// the integration cloud tests are run.
```
Comment (tychoish) [inline cloud/new_ec2_integration_test.go]: is there any rounding that might impact things? 

where are the credentials pulled from.

File: cloud/new_ec2.go (hunk 1 above)
Comment (bsamek) [reply cloud/new_ec2.go]: > we don't get the reason why we're removing the intent host here, (either err, or reservation,) so this is only logged if we have trouble with the database connection.

Fixed.

> also cloud provider and host provider are probably always mostly the same thing? or at least cloud provider won't add much information.

Host is not always available where I use `cloud_provider`, like in the client wrappers. By adding this field, we can search Splunk for everything this provider is up to, which is useful. I made it a constant though, which it should have been.

> does h.Remove() set h to nil?

It removes it from the collection, but it does not set h to nil.

File: cloud/new_ec2.go (hunk 2 above)
Comment (bsamek) [reply cloud/new_ec2.go]: Explained above.

File: cloud/new_ec2_integration_test.go (hunk 3 above)
Comment (bsamek) [reply cloud/new_ec2_integration_test.go]: > is there any rounding that might impact things?

I don't think so.

> where are the credentials pulled from.

From the settings file, like any other integration test.

**Pull request merged**